osu.Game/Screens/Edit/Compose/Components/SelectionBoxRotationHandle.cs

```diff
@@ -1,19 +1,34 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using osu.Framework.Allocation;
+using osu.Framework.Bindables;
 using osu.Framework.Extensions.EnumExtensions;
+using osu.Framework.Extensions.LocalisationExtensions;
 using osu.Framework.Graphics;
+using osu.Framework.Graphics.Cursor;
 using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Framework.Localisation;
 using osuTK;
 using osuTK.Graphics;
 
 namespace osu.Game.Screens.Edit.Compose.Components
 {
-    public class SelectionBoxRotationHandle : SelectionBoxDragHandle
+    public class SelectionBoxRotationHandle : SelectionBoxDragHandle, IHasTooltip
     {
+        public Action<float> HandleRotate { get; set; }
+
+        public LocalisableString TooltipText { get; private set; }
+
         private SpriteIcon icon;
 
+        private readonly Bindable<float?> cumulativeRotation = new Bindable<float?>();
+
+        [Resolved]
+        private SelectionBox selectionBox { get; set; }
```

Comment:
Usually we avoid DI one level above (and pass in via ctor or property instead). Not sure it's worth changing here, but mentioning it as a general guideline.